Cesium.CodeGen/Ir/Types/StructType.cs

```diff
@@ -117,14 +119,21 @@ public TypeReference Resolve(TranslationUnitContext context)
 
         if (resolved == null)
         {
-            if (IsAnon)
+            try_again:
+            if (IsAnon || IsLocal)
             {
                 if (AnonType == null)
                     EmitAsAnonStructure(context);
 
                 return AnonType!; // not null
             }
-            throw new CompilationException($"Type {this} was not found.");
+
+            if (Members.Count == 0)
+                throw new CompilationException($"Can't find the type with name: {Identifier}");
+
+            // maybe its Local type?
+            IsLocal = true;
+            goto try_again;
```

Comment:
This is so messy. Could it be improved with a local function or something? Essentially, if I understand correctly, you just want to call the `AnonType` check twice: if the type is anonymous, or if it has zero members.

You don't need an endless loop modelled via `goto` or even a mutable member to do this.